base/shell/explorer/explorer.cpp

```diff
@@ -49,6 +49,41 @@ static VOID InitializeAtlModule(HINSTANCE hInstance, BOOL bInitialize)
     }
 }
 
+static BOOL
+GetServerAdminUIDefault()
+{
+    BOOL server;
+    DWORD value = 0, size = sizeof(value);
+    LPCWSTR rosregpath = L"SYSTEM\\CurrentControlSet\\Control\\ReactOS\\Settings\\Version";
+    if (SHGetValueW(HKEY_LOCAL_MACHINE, rosregpath, L"ReportAsWorkstation", NULL, &value, &size) == NO_ERROR)
+        server = value == 0;
+    else
+        server = IsOS(OS_ANYSERVER);
+    return server && IsUserAnAdmin();
```

Comment:
> Since the `ReportAsWorkstation` only does change `wProductType` to `VER_NT_WORKSTATION`, you can simply use `RtlGetVersion`

So, there is a strange bug that caused me not to do this.

If you choose "server" during ROS install, the Workstation checkbox in sysdm.cpl works correctly and changes `wProductType`.

However, if you choose "workstation" during ROS install, the checkbox in sysdm.cpl does nothing! You cannot get `wProductType` to report server.

I will fix this first and then come back and update this PR.